src/palace.pyx

```diff
@@ -121,10 +121,17 @@ def query_extension(name: str) -> bool:
     return devmgr.query_extension(name)
 
 
-def current_context() -> Optional[Context]:
-    """Return the context that is currently used."""
+def current_context(thread: bool = False) -> Optional[Context]:
+    """Return the context that is currently used.
+
+    If `thread` is set to `True`,
+    it will return the current context thread.
```

Comment:
Something is wrong with the phrasing here.  Please see alure's doc and also document the exception might be raised.